opensearch-project/OpenSearch-Dashboards
# [Workspace]Import sample data to current workspace

### Description

For now, the import sample data page is a sub page under home application. The home application is not visible inside specific workspace. This PR separate import sample data page and register it as a new standalone application. Then import sample data page can be used inside workspace. This PR added a new entry point for import sample data in workspace overview page. User can visit import sample data page by this link quickly. 

### Issues Resolved

#6106 

## Screenshot
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/4034161/dd81c981-831a-4285-a7dc-26020d8acade)

![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/4034161/5499c406-32f6-47ac-a745-ad38793b2e90)

![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/4034161/9fa95902-ff71-4728-b481-4a7557dce81f)


The sample data can be imported to specific workspace.

## Testing the changes

Since switch workspace by manually wasn't support in current main branch. We need to apply some patch to testing all the changes. 
1. Checkout the branch code and run `yarn osd bootstrap`
2. enable workspace and start OSD server
- add `workspace.enabled: true` and `savedObjects.permission.enabled: true` to the `config/opensearch_dashboards.yml` to enable workspace feature flag
- run `yarn start --no-base-path` start OSD server to make sure no random base path will be appended
3. Navigate to [workspace creator page](http://localhost:5601/app/workspace_create) and create a test workspace
4. Click left dropdown switch to created workspace
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/4034161/9397e8f4-c6ad-47a4-a198-7aad609bd1d2)

5. Click "Discover pre-loaded datasets before adding your own."  to visit workspace's Add sample data page
<img width="1723" alt="image" src="https://github.com/opensearch-project/OpenSearch-Dashboards/assets/4034161/7713f1e8-1967-4bd5-a2d3-da268543ed85">


6. Click all `Add data` button, all sample data will be imported like below image

![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/4034161/9fa95902-ff71-4728-b481-4a7557dce81f)
    There will be four dashboards in current workspace.

7. Click `View data` to view imported dashboard like below image
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/4034161/4da4e86a-3aaf-4a4c-9425-cc49dd39832c)


8. Create another workspace, and visit another workspace's add sample data page like step 4 & 5.
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/4034161/ab4dee8d-45d5-481a-aef4-40d5d248b2b6)

The sample data was not import in the new workspace.

9. Switch back to the first workspace, click remove button, the sample data will be removed like below image 
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/4034161/78078bf2-8c89-4cef-838f-324edfeea251)
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/4034161/16497190-1565-4039-8ade-5a60f75f0404)
The `[Flights] Global Flight Dashboard` will be removed in this workspace like above image.

## Changelog
- feat: [Workspace]Import sample data to current workspace

### Check List
